src/bp/core/services/kvs/index.ts

```diff
@@ -19,9 +21,11 @@ export class KeyValueStore {
     private logger: Logger
   ) {}
 
-  upsert = (botId: string, key: string, value) => {
+  private upsert = (botId: string | undefined, key: string, value) => {
```

Comment:
When you have a mix of public and private methods, you can add _ before the private method name to differentiate them easily.

Also, you could write your method call like that (it will use global by default if undefined)
```js
private _upsert = (botId: string = GLOBAL, key: ...)
```